lib/config/config-file.js

```diff
@@ -0,0 +1,85 @@
+const path = require('path')
+const fs = require('fs')
+const _ = require('lodash')
+const cosmiconfig = require('cosmiconfig')
+const { ConfigMissingError } = require('../common/errors')
+const packageJson = require('../../package.json')
+
+const getSolhintCoreConfigPath = name => {
+  if (name === 'solhint:recommended') {
+    return path.resolve(__dirname, '../../conf/rulesets/solhint-recommended.js')
+  }
+
+  if (name === 'solhint:all') {
+    return path.resolve(__dirname, '../../conf/rulesets/solhint-all.js')
+  }
+
+  if (name === 'solhint:default') {
+    return path.resolve(__dirname, '../../conf/rulesets/solhint-default.js')
+  }
+
+  throw new ConfigMissingError(name)
+}
+
+const createEmptyConfig = () => ({
+  excludedFiles: {},
+  extends: {},
+  globals: {},
+  env: {},
+  rules: {},
+  parserOptions: {}
+})
+
+const loadConfig = () => {
+  // Use cosmiconfig to get the config from different sources
+  const appDirectory = fs.realpathSync(process.cwd())
+  const moduleName = packageJson.name
+  const cosmiconfigOptions = {
+    searchPlaces: [
+      'package.json',
+      `.${moduleName}.json`,
+      `.${moduleName}rc`,
+      `.${moduleName}rc.json`,
+      `.${moduleName}rc.yaml`,
+      `.${moduleName}rc.yml`,
+      `.${moduleName}rc.js`,
+      `${moduleName}.config.js`
+    ]
+  }
+
+  const explorer = cosmiconfig(moduleName, cosmiconfigOptions)
+  const searchedFor = explorer.searchSync(appDirectory)
+  return searchedFor.config || createEmptyConfig()
+}
+
+const applyExtends = config => {
+  let configExtends = config.extends
+
+  // normalize into an array for easier handling
+  if (!Array.isArray(config.extends)) {
+    configExtends = [config.extends]
+  }
+
+  return configExtends.reduceRight((previousValue, parentPath) => {
```

Comment:
Any reason to reduce from the right? This means that the rules that appear first in the array will override the ones that appear next. I think it makes more sense to do it the other way around, and that's the way eslint does it if I'm not wrong.